docs/source/secured_private_asset_transfer_tutorial.rst

```diff
@@ -0,0 +1,346 @@
+
+Agreeing to the transfer
+---------------------------------------
+
+After a asset is created, channel members can use the smart contract to agree to transfer the asset:
+
+- The owner of the asset can change the description in the public ownership record, for example to advertise that the asset is for sale. Smart contract access control enforces that this change needs to be submitted from a member of the asset owner organization. The state based endorsement policy enforces that this description change must be endorsed by a peer from the owner's organization.
+
+The asset owner and the asset buyer agree to transfer the asset for a certain price:
+- The price agreed to by the buyer and the seller is stored in each organization's implicit private data collection. The private data collection keeps the agreed price secret from other members of the channel. The endorsement policy of the private data collection ensures that the respective organization's peer endorsed the price agreement, and the smart contract access control logic ensures that the price agreement was submitted by a client of the associated organization.
+- A hash of each price agreement is stored on the ledger. The two hashes will match only if the two organizations have agreed to the same price. This allows the organizations to verify that they have come to agreement on the transfer details before the transfer takes place. A random trade id is added to the price agreement, which serves as a *salt* to ensure that other channel members can not use the hash on the ledger to guess the price.
+
+Transferring the asset
+---------------------------------------
+
+After the two organizations have agreed to the same price, the asset owner can use the transfer function to transfer the asset to the buyer:
+
+- Smart contract access control ensures that the transfer must be initiated by a member of the organization that owns the asset.
+- The transfer function verifies that the asset's private immutable properties passed to the transfer function matches the on chain hash if the asset data in private collection, to ensure that the asset owner is *selling* the same asset that they own.
+- The transfer function uses the hash of the price agreement on the ledger to ensure that both organizations have agreed to the same price.
+- If the transfer conditions are met, the transfer function adds the asset to the implicit private data collection of the buyer, and deletes the asset from the collection of the seller. The transfer also updates the owner in the public ownership record.
+- Because of the endorsement policies of the seller and buyer implicit data collections, and the state based endorsement policy of the public record (requiring the seller to endorse), the transfer needs to be endorsed by peers from both buyer and seller.
+- The state based endorsement policy of the public asset record is updated so that only a peer of the new owner of the asset can update or sell their new asset.
+
```

Comment:
- The price agreements are also deleted from both the seller and buyer implicit private data collection, and a sales receipt is created in each private data collection.